{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from atom3d.datasets import load_dataset\n",
    "from tqdm import tqdm\n",
    "import collections as col\n",
    "from collapse.utils import pdb_from_fname, deserialize\n",
    "from collapse.atom_info import aa_to_letter\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "label_dict = {'LP/P': 'pathogenic', 'LB/B': 'benign', 'US': 'unknown'}\n",
    "mut_data = []\n",
    "with open('data/humsavar.txt') as f:\n",
    "    for i in range(44):\n",
    "        next(f)\n",
    "    for l in f:\n",
    "        gene, uniprot, _, variant, lab, snp, dz = l[:12].strip(), l[12:23].strip(), l[23:35].strip(), l[35:50].strip(), l[50:59].strip(), l[59:74].strip(), l[74:].strip()\n",
    "        if lab not in label_dict:\n",
    "            continue\n",
    "        label = label_dict[lab]\n",
    "        wt = variant[2:5]\n",
    "        mut = variant[-3:]\n",
    "        pos = int(variant[5:-3])\n",
    "        mut_data.append([uniprot, wt, pos, mut, label, dz])\n",
    "mut_data = pd.DataFrame(mut_data, columns=['uniprot', 'wt', 'pos', 'mut', 'label', 'disease'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprot_rows = mut_data.groupby('uniprot').groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot</th>\n",
       "      <th>wt</th>\n",
       "      <th>pos</th>\n",
       "      <th>mut</th>\n",
       "      <th>label</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P04217</td>\n",
       "      <td>His</td>\n",
       "      <td>52</td>\n",
       "      <td>Arg</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P04217</td>\n",
       "      <td>His</td>\n",
       "      <td>395</td>\n",
       "      <td>Arg</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q9NQ94</td>\n",
       "      <td>Val</td>\n",
       "      <td>555</td>\n",
       "      <td>Met</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Q9NQ94</td>\n",
       "      <td>Ala</td>\n",
       "      <td>558</td>\n",
       "      <td>Ser</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P01023</td>\n",
       "      <td>Arg</td>\n",
       "      <td>704</td>\n",
       "      <td>His</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82407</th>\n",
       "      <td>Q8N402</td>\n",
       "      <td>Thr</td>\n",
       "      <td>131</td>\n",
       "      <td>Pro</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82408</th>\n",
       "      <td>Q96M66</td>\n",
       "      <td>Arg</td>\n",
       "      <td>37</td>\n",
       "      <td>His</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82409</th>\n",
       "      <td>Q96M66</td>\n",
       "      <td>Arg</td>\n",
       "      <td>171</td>\n",
       "      <td>Ser</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82410</th>\n",
       "      <td>Q9N2K0</td>\n",
       "      <td>Val</td>\n",
       "      <td>81</td>\n",
       "      <td>Leu</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82411</th>\n",
       "      <td>Q9N2K0</td>\n",
       "      <td>Phe</td>\n",
       "      <td>150</td>\n",
       "      <td>Leu</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>82412 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      uniprot   wt  pos  mut   label disease\n",
       "0      P04217  His   52  Arg  benign       -\n",
       "1      P04217  His  395  Arg  benign       -\n",
       "2      Q9NQ94  Val  555  Met  benign       -\n",
       "3      Q9NQ94  Ala  558  Ser  benign       -\n",
       "4      P01023  Arg  704  His  benign       -\n",
       "...       ...  ...  ...  ...     ...     ...\n",
       "82407  Q8N402  Thr  131  Pro  benign       -\n",
       "82408  Q96M66  Arg   37  His  benign       -\n",
       "82409  Q96M66  Arg  171  Ser  benign       -\n",
       "82410  Q9N2K0  Val   81  Leu  benign       -\n",
       "82411  Q9N2K0  Phe  150  Leu  benign       -\n",
       "\n",
       "[82412 rows x 6 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "label\n",
       "benign        39654\n",
       "pathogenic    32627\n",
       "unknown       10131\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_data.value_counts('label')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13055"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_data.uniprot.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "k = 50000\n",
    "clusterer = deserialize(f'data/pdb100_cluster_fit_{k}.pkl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21834"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_proteome = load_dataset('/scratch/users/aderry/collapse/datasets/af2_human_proteome/full', 'lmdb')\n",
    "len(human_proteome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 21834/21834 [3:31:24<00:00,  1.72it/s]   \n"
     ]
    }
   ],
   "source": [
    "cluster_cts = np.zeros(k)\n",
    "cluster_mutations = col.defaultdict(list)\n",
    "uniprot_clusters = {}\n",
    "cluster_refs = col.defaultdict(list)\n",
    "for item in tqdm(human_proteome):\n",
    "    if item is None:\n",
    "        continue\n",
    "    afid, _ = pdb_from_fname(item['id'])\n",
    "    \n",
    "    high_conf_idx = np.array(item['confidence']) >= 70\n",
    "    if np.sum(high_conf_idx) == 0:\n",
    "        continue\n",
    "    af_resids = np.array(item['resids'])[high_conf_idx]\n",
    "    embeddings = item['embeddings'][high_conf_idx]\n",
    "    \n",
    "    clusters = clusterer.predict(embeddings)\n",
    "    uniprot_clusters[afid] = af_resids, clusters\n",
    "    np.add.at(cluster_cts, clusters, 1)\n",
    "    if afid in uniprot_rows:\n",
    "        wt = mut_data.loc[uniprot_rows[afid], 'wt'].values\n",
    "        pos = mut_data.loc[uniprot_rows[afid], 'pos'].values\n",
    "        mut = mut_data.loc[uniprot_rows[afid], 'mut'].values\n",
    "        labels = mut_data.loc[uniprot_rows[afid], 'label'].values\n",
    "        resids = [aa_to_letter(r.upper()) + str(p) for r,p in zip(wt, pos)]\n",
    "        resids_to_labels = dict(zip(resids, labels))\n",
    "        var_idx = np.isin(af_resids, resids)\n",
    "        var_resids = af_resids[var_idx]\n",
    "        var_labels = [resids_to_labels[v] for v in var_resids]\n",
    "        var_clusters = clusters[var_idx]\n",
    "        for i in range(len(var_clusters)):\n",
    "            cluster_mutations[var_clusters[i]].append(var_labels[i])\n",
    "            cluster_refs[var_clusters[i]].append(afid + '_' + var_resids[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collapse.utils import serialize\n",
    "serialize(cluster_cts, f'data/af2_human_cluster_cts_{k}.pkl')\n",
    "serialize(cluster_mutations, f'data/af2_human_cluster_variants_{k}.pkl')\n",
    "serialize(uniprot_clusters, f'data/af2_human_cluster_resids_{k}.pkl')\n",
    "serialize(cluster_refs, f'data/af2_human_cluster_variant_refs_{k}.pkl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "k = 50000\n",
    "cluster_cts = deserialize(f'data/af2_human_cluster_cts_{k}.pkl')\n",
    "cluster_mutations = deserialize(f'data/af2_human_cluster_variants_{k}.pkl')\n",
    "uniprot_clusters = deserialize(f'data/af2_human_cluster_resids_{k}.pkl')\n",
    "cluster_refs = deserialize(f'data/af2_human_cluster_variant_refs_{k}.pkl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "23075"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(cluster_mutations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2980.0"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_cts[24491]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['benign',\n",
       " 'pathogenic',\n",
       " 'pathogenic',\n",
       " 'pathogenic',\n",
       " 'pathogenic',\n",
       " 'pathogenic',\n",
       " 'pathogenic',\n",
       " 'pathogenic',\n",
       " 'benign',\n",
       " 'unknown',\n",
       " 'benign',\n",
       " 'pathogenic']"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_mutations[24491]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'benign': 20377, 'pathogenic': 22571, 'unknown': 5419}"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "label_cts = {'benign': 0, 'pathogenic': 0, 'unknown': 0}\n",
    "for _,v in cluster_mutations.items():\n",
    "    ct = col.Counter(v)\n",
    "    for k in ct:\n",
    "        label_cts[k] += ct[k]\n",
    "label_cts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.0\n"
     ]
    }
   ],
   "source": [
    "M = np.sum(list(label_cts.values()))\n",
    "n = label_cts['pathogenic']\n",
    "N = cluster_cts[24491]\n",
    "k = col.Counter(cluster_mutations[24491])['pathogenic']\n",
    "pval = hypergeom.sf(k, M, n, N)\n",
    "print(pval)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import hypergeom"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "M = np.sum(list(label_cts.values()))\n",
    "n = label_cts['pathogenic']\n",
    "\n",
    "cluster_path_pvals = {}\n",
    "for clust, muts in cluster_mutations.items():\n",
    "    N = len(muts)\n",
    "    k = col.Counter(muts)['pathogenic']\n",
    "    pval = hypergeom.sf(k, M, n, N) \n",
    "    pval_corr = pval * len(cluster_mutations)\n",
    "    cluster_path_pvals[clust] = [pval, pval_corr, muts, float(k) / N]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "path_enrich = pd.DataFrame(cluster_path_pvals).T\n",
    "path_enrich.columns = ['pval', 'pval_corr', 'mutations', 'pct_pathogenic']\n",
    "path_enrich['n_mut'] = path_enrich.mutations.apply(len)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pval</th>\n",
       "      <th>pval_corr</th>\n",
       "      <th>mutations</th>\n",
       "      <th>pct_pathogenic</th>\n",
       "      <th>n_mut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7043</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31037</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23310</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39542</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48393</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31407</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48563</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8771</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47510</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25011</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4095</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38836</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31025</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18585</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44777</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44065</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14659</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33252</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      pval pval_corr                                          mutations  \\\n",
       "7043   0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "31037  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "23310  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "39542  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "48393  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "31407  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "160    0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "48563  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "8771   0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "47510  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "25011  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "4095   0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "38836  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "31025  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "184    0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "18585  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "44777  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "44065  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "14659  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "33252  0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "\n",
       "      pct_pathogenic  n_mut  \n",
       "7043             1.0     19  \n",
       "31037            1.0      9  \n",
       "23310            1.0      9  \n",
       "39542            1.0      8  \n",
       "48393            1.0      8  \n",
       "31407            1.0      8  \n",
       "160              1.0      7  \n",
       "48563            1.0      7  \n",
       "8771             1.0      7  \n",
       "47510            1.0      7  \n",
       "25011            1.0      7  \n",
       "4095             1.0      7  \n",
       "38836            1.0      7  \n",
       "31025            1.0      7  \n",
       "184              1.0      7  \n",
       "18585            1.0      7  \n",
       "44777            1.0      6  \n",
       "44065            1.0      6  \n",
       "14659            1.0      6  \n",
       "33252            1.0      6  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path_enrich.sort_values(['pval_corr', 'n_mut'], ascending=[True, False]).head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAATEAAADZCAYAAABIBQczAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy81sbWrAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAoZElEQVR4nO3deVhU9f4H8De7MIhJiQsSLo9AouioLAY4liym6Q0MytREhDSK3FJzAQtNRVMKMajrVpJmgkulocWVQMnIMr0qgmHIIqAGyjIyzDDf3x/+mMsIyhyYfT6v5+F5mHPOnPOegfnM93zPOd9jxBhjIIQQHWWs6QCEENIVVMQIITqNihghRKdRESOE6DQqYoQQnUZFjBCi06iIEUJ0GhUxQohOM9V0AE2prq7G6dOn0b9/f1hYWGg6DiGkFZFIhLKyMvj4+MDW1vaxy+p0Efvjjz8wffp0/Pbbb7CxseH03NOnT2Pp0qUqSkYIUYbNmzdj6tSpj11Gp4vYF198gWHDhnXquf379wfw4E0aPHiwMmMRQrqoqKgIS5culX1OH0dni1hubi6GDx+O6urqTj2/ZRdy8ODBcHV1VWY0QoiSKNLVo9GO/dTUVAQHB2PYsGFYtGiR3Lza2losWLAAfD4fPj4+2LNnj9z8L7/8EjNmzFBjWkKINtJoS8zOzg5RUVHIzc1FTU2N3Ly4uDg0NTUhJycH5eXlCAsLw8CBAyEQCJCZmQlPT09YWlpqKDkhRFtotCUWEBAAPz8/9OzZU266UChERkYGFi1aBGtrazg7OyM0NBTp6ekAgMLCQmRlZWHu3LkoKCjA8uXLNRGfEKIFtLJPrLi4GIwxODk5yaa5uLjg5MmTAIA333wTb775JgBg1qxZiI+P10hOQojmaWUREwqFsLa2lptmY2ODhoaGNsvu3btX5XnEEinMTI07nEYIUT+tLGJWVlZtClZdXR14PJ5G8piZGiMs7gSEjRIAgFU3U+yJDdRIFkKIPK1sSgwYMAAAcO3aNdm0/Px8DBkyREOJAGGjBPdFD35aihkhRPM0WsQkEglEIhEkEgmkUilEIhHEYjGsrKwQGBiIhIQE1NfXo7CwEGlpaZg2bZom4xJCtJBGi1hycjLc3NyQkpKCjIwMuLm5ISYmBgCwZs0amJqawtfXF+Hh4YiMjIRAINBkXEIIR2KJVKFpXaHRPrHo6GhER0e3O8/GxgaJiYlqTkQIUSZ19CdrZcc+IUR/tPQnq4pWduwTQoiiOBex4uJi2UXXQqEQH3/8MZKSkiASiZQejhBCOsK5iC1ZsgS3bt0CAHz88cf46aef8OOPP2Ljxo1KD0cIIR3hXMRKS0tllwOdPHkSycnJ2LVrF3766SelhyOEkI5w7thnjMHIyAilpaUwNjaGg4MDAKC+vl7p4QghpCOci5iLiwuSk5NRUVEBb29vAEBVVVWbax0JIUQdOO9Orlq1Cjk5OSgpKUFUVBSAB6OsthQ0QghRJ04tMYlEguvXr2PPnj1yw8YGBQUhKChI6eEIIaQjnFpipqamWLVqFd3ijBCiNTjvTjo7O+P69euqyEIIIZxx7tifOHEi3n77bcyZMwf29vYwMjKSzRs7dqxSwxFCSEc4F7GWk1pbRptoYWRkhPz8fOWkIoQQBXEuYlevXlVFDkK67OEhw2kIccNAo1gQvdF62BcaQtxwcP6aam5uxmeffYaAgACMHj0aAJCTk4NvvvlG6eEI4apl2BcaQtxwcC5iSUlJ+OGHHxAdHS3r1Hd0dMT+/fuVHo4QQjrCuYgdPXoUycnJmDJlCoyNHzy9f//+KC8vV3o4QgjpCOci1tDQgD59+shNa25uhomJidJCEUKIojgXMScnJ2RkZMhNy8zMxNChQ5UWihBCFMX56OS7776LsLAw/PTTTxCJRFi1ahVOnDiBXbt2qSIfIYQ8FueW2IgRI5Ceno6ePXvCw8MDUqkUu3fvhpubmyryEULIY3FuiVVWVmLQoEFYvXp1m+kP95WpSkVFBRYtWgQzMzMYGRlh8+bN6N27t1q2TQjRLpxbYpMmTWp3+tSpU7scRlF2dnbYt28f9u7di5deeonOUSPEgHVqeOqHSaXKvaNvR1ofCW1oaMCQIUPUun1CiPZQuIgtW7YMACAWi2W/tygpKcGgQYM4bzw1NRWHDh1CYWEh/P39kZCQIJtXW1uLmJgYZGdng8fjISIiAmFhYbL5Fy9exNq1a1FbW4sdO3Zw3jYhRD8oXMRat35a/25kZARPT0+EhoZy3ridnR2ioqKQm5uLmpoauXlxcXFoampCTk4OysvLERYWhoEDB0IgEAAA3NzccPDgQRw/fhwpKSn48MMPOW+fEKL7FC5iGzZsAAAMGDAA8+bNU8rGAwICAAD5+flyRUwoFCIjIwOHDh2CtbU1nJ2dERoaivT0dAgEAjQ1NcHc3BwA0L17d1haWiolDyFE93DuE3vttdfQ2NiIbt26QSqVIj09HWZmZnjppZeUFqq4uBiMMdn9LYEHd1k6efIkAODPP//EJ598AmNjY5ibm2P9+vVK2zYhRLdwLmLz5s3D8uXLMWLECGzfvh1ff/01TExMUFxcjIULFyollFAobHMLOBsbGzQ0NAAAPDw88NVXXyllW4QQ3cb5FIuioiK4uroCAL7//nvs3LkTX331Fb799lulhbKyspIVrBZ1dXXg8XhK2wbpPLFE+tjHhKgT55aYVCqFqakpqqqq0NDQABcXFwBo0zHfFQMGDAAAXLt2TXb6RH5+Pp1KoSVo8MHHoxFm1YtzEXv66adx+PBhlJSUwNPTEwBQXV0NKysrzhuXSCRobm6GRCKBVCqFSCSCsbExrKysEBgYiISEBGzatAk3b95EWloa9X1pkZbBB0lbVOTVi3MRW7p0KZYtWwZzc3MkJycDALKysjBs2DDOG09OTkZSUpLscUZGBoKCgrBx40asWbMGq1evhq+vL3g8HiIjI2WnVxCi7ajIqw/nIubl5YXs7Gy5aVOmTMGUKVM4bzw6OhrR0dHtzrOxsUFiYiLndRJCDEun+sQeRgMiEkI0hXMRGzp0qNwNc1uj+04SQtSNcxH78ssv5R5XVVVh9+7dCAkJUVooQghRFOci5uHh0WbayJEjsWzZMkyfPl0poQghRFFKOXmlX79+KCgoUMaqCCGEE84tsdLSUrnHQqEQBw8ehL29vdJCEUKIojgXMX9/f7mOfcYYHBwcEB8fr9RghBCiCM5FLDMzU+4xj8fDE088oaw8hBDCCeciRruNhBBtolAR++STTxRa2YIFC7oUhhBCuFKoiJ07d67DZR51AiwhhKiSQkVs7969qs5B1ISGiSH6hnOfWFVVFSwsLOQ68+/evQuRSEQ3sNUBNEwM0Tecv4IXLFiAsrIyuWllZWXUH6ZDWoaJETbSUDHaikbPVRznlljr4albDB06FNevX1daKEIMHbWYFce5JWZhYQGhUCg37f79+zA2pn4VQpSJWsyK4Vx5hg8fjn//+99y03bu3Ak3NzelhSKEEEVx3p189913MWPGDGRmZmLgwIG4ceMGKisr6RZqhBCN4FzEBg8ejGPHjuHo0aMoKyvDiBEj8NJLL+HJJ59URT5CCHkszkUMAJ588kmEh4crOwshhHBGvfGEEJ1GRYwQotOoiBGNoJM5ibIoVMS2bNki+/33339XWRhF5eXlITQ0FDNmzMDixYshFos1HYlw1HIyZ+jKYwiLO0HXb3ZRe18ChvLFoNB/zr59+2S/R0ZGqiyMohwdHbF371589dVXsLe3x8mTJzUdiXSCLpzMqSstxtZfCob2xaDQ0clevXph9+7dcHFxQXNzM86ePQvGWJvlxo4dq/SA7Wl9obm5uTkNA9QKjVKhXLp0+U/Ll4KhUaiIxcTEYO3atSgtLYVUKkVYWFibZYyMjDjfPDc1NRWHDh1CYWEh/P39kZCQIJtXW1uLmJgYZGdng8fjISIios12S0tLcfr0acybN4/TdvWZLn3odIWhFgddoVAR8/b2RkZGBgCAz+fj/PnzStm4nZ0doqKikJubi5qaGrl5cXFxaGpqQk5ODsrLyxEWFoaBAwdCIBAAAO7du4elS5diw4YNMDc3V0oefUEfOmJIOO9nHD58WGkbDwgIgJ+fH3r27Ck3XSgUIiMjA4sWLYK1tTWcnZ0RGhqK9PR0AEBTUxMWLlyIxYsXY9CgQUrLQwjRPZzP2B8wYADy8/Nx4MABVFRUoG/fvggNDcXQoUOVFqq4uBiMMTg5Ocmmubi4yDrw09LSkJ+fj23btgEAQkNDMWXKFKVtnxCiOzgXsezsbERFRWHcuHEYNGgQSkpK8MorryApKUm2q9dVQqEQ1tbWctNsbGzQ0NAAAHjttdfw2muvKWVbhBDdxrmIJSYmYtOmTZg0aZJs2vHjx5GYmKi0ImZlZSUrWC3q6urA4/GUsn5CiP7g3Cd248YNTJw4UW5aYGAgbty4obRQAwYMAABcu3ZNNi0/Px9DhgxR2jYIIfqBcxGztbVFYWGh3LRr167B1taW88YlEglEIhEkEgmkUilEIhHEYjGsrKwQGBiIhIQE1NfXo7CwEGlpaZg2bRrnbRBC9Bvn3cmQkBDMnz8fERERcHBwQElJCXbt2tWpPqrk5GQkJSXJHmdkZCAoKAgbN27EmjVrsHr1avj6+oLH4yEyMlJpu6uEEP3BuYiFh4fD2NgYqampqKysRJ8+fTBz5kzMmTOH88ajo6MRHR3d7jwbGxskJiZyXichxLBwLmLGxsYIDw+nQREJIVqBLqojhOg0KmI6RFdGVCBEnTo1xj7RDHVc3N3eqBc0EgbRZlTEdIyqL+5uXSgB0EgYROtx+noVi8UYPXo0RCKRqvIQLdBSKLV9wEJl0NdddDNTY719bQ/j1BIzMzNDjx49IJXq55tBDI8mxl9Tx8CVpibGBjO2HOd3bu7cudiwYQOamppUkYcYIE23GtQ9TLY67y+gC0OAdxXnPrHPP/8cd+7cwaFDh2Brawtj4//9AbKyspSZjRgIQ2o1tKCBK5WHcxFbuHChCmIQQh9s0jmci1hQUJAqchA9QqdpEHXq1CkWpaWlOHbsGG7duoXY2FjcuHEDEokEgwcPVnY+ooPoNA3t09LvqI93wuL8Cn755RdMnToVeXl5svH2b9++jfj4eKWHI7pLH0/TaO8ABKAbpy607nfUt/tScm6JffTRR9i8eTP8/Pzg7u4OABg2bBiuXLmi9HCEaJOHD0AAutfK1Md+R85F7MaNG/Dz8wMA2U1ru3XrRifAEoOhj4VAl3FuT9rZ2bUZirqoqAh9+vRRWihCCFEU5yL28ssvY+HChThz5gykUinOnTuHVatWITQ0VBX5DJou9LXoEno/9RPn3cmwsDA0NDRgwYIFqK+vR0REBF599VXMnDlTFfkMWuv+F1sbC6S858f5+fpyBEoZHu7P6sx7SrRPp0Z2bRlW+p9//kH37t1hbm6uimwErY/ycT8b5uGOaF350D6q+CqjILfuz+rMe0q0T6f+ilKpFOfPn5eNsc/n8+UuPyLapSuFUBP04SggUR/O/9WlpaWYP38+iouL0bNnT9TU1MDR0REpKSl4+umnVZGRGCg6CkgUwbn5FBsbCz6fj7y8PJw+fRp5eXkYM2YMYmJiVJGPEKICmh45RJk4t8T++9//IiUlBRYWFgAAHo+HFStWwMfHR+nhHuXu3buIiIhAUVER9u3bh2eeeUZt2yZEH+jTyCGcW2K9e/fG3bt35abV1taid+/eysrUIR6Ph88//xyBgbr5phPD8ahLlbq6DmW1mvRhvDGFWmKlpaWy38PCwhAdHY233noL9vb2KC8vR3JycqdunttZZmZmsLW1Vdv21IVOh9A/yjhCrE+tJlVQqIj5+/vLLjFijAEA5s2bJ7fMhQsXEBISwmnjqampOHToEAoLC+Hv74+EhATZvNraWsTExCA7Oxs8Hg8REREICwvjtH5t1l7B0sXTIYhilHGEmA50tE+hdzQzM1MlG7ezs0NUVBRyc3NRU1MjNy8uLg5NTU3IyclBeXk5wsLCMHDgQAgEApVkUbdHnXipa6dDEKJpCn1S7O3tVbLxgIAAAEB+fr5cERMKhcjIyMChQ4dgbW0NZ2dnhIaGIj09XW+KGEAnXhKiDJ365Fy4cAEXL15EQ0OD3PT58+crJVRxcTEYY3BycpJNc3FxwcmTJ2WPw8LC8Ndff+H69euYNm0aXnnlFaVsmxCiWzgXsW3btiElJQWDBg2ClZWVbLqRkZHSiphQKIS1tbXcNBsbG7miuWfPHqVsy5Apcm2lIV1/qYnXakjvr6pwLmL79u3Dvn37MGLECFXkAQBYWVm1aeXV1dWBx+OpbJuGSJEjZ7p6/WVnaOK1tneJlT6/x6rAufybmJhg+PDhqsgiM2DAAADAtWvXZNPy8/MxZMgQlW7XUP3vYEJzl5bRF5p4ra2H8zaE91iZOBexkJAQ7N+/Xykbl0gkEIlEkEgkkEqlEIlEEIvFsLKyQmBgIBISElBfX4/CwkKkpaVh2rRpStkuIUR/cN6dDA8PR0hICFJTU9GrVy+5eV9++SWndSUnJyMpKUn2OCMjA0FBQdi4cSPWrFmD1atXw9fXFzweD5GRkXp1ZLKrDK0vRZ/v1qMNVDn8kapxLmLLli0DYwwCgQCWlpZd2njLuGTtsbGxQWJiYpfWr88MrS+FzlpXLV0e/ohzETt79iwyMzP18rIfXWRo55rRWeuqpYvvb6duFNLVFhghRPvpynA9nIvYggUL8MEHH6CqqgpSqVTuhxCiP3Tlhruc9z8WL14MADh69Gibefn5+V1PRHSKoR1gMETavovJuYhxPQJJ9JshnQxLtBPnIubh4aGKHETH0egbRFM4/8f98ssvj5w3duzYLoUhhBCuOBex9kZwbRkwkfrECCHqxrmIXb16Ve5xVVUVtm7dCn9/f6WFIoQQRXX5kFLv3r0RExODzZs3KyMPIYRwopTj4owx3L59WxmrIoQQTjjvTqalpck9vn//Po4dOwY+n6+0UPqIzqVSL30+f02fX1tncC5in376qdxjHo+HYcOGYeHChcrKpJfoXCr10ufz1/T5tXUG5yL2n//8RxU5DAKdS6V++vye6/Nr44Lao4QQnaZwCV+xYsVj5xsZGWH9+vVdDkQIIVwo3BKTSCSP/MnKysLhw4dVmZMQQtqlcEusvfPA8vPz8eGHH0IqlWLlypVKDUaILjCkI4XtDxHeDDNTk0c+VodO9QhWV1cjISEBR48eRUhICJKSkvDEE08oORoh2s+QjhQ+6rV29FjlubgsLJFI8OWXX+LTTz/FsGHDkJ6eTrdRIwSGdaTw4dfa0WNVU3grWVlZ2LhxIxhjiI+Px4QJE1SZixBCFKJwEZs/fz5sbW0RFBSES5cu4dKlS22WWbBggVLDEUJIRxQuYu7u7gCAixcvtju/ZTgedUlISEBeXh569eqF+Ph4unkJIQZK4SK2d+9eVebgpKCgANevX8f+/fuxd+9epKWlYdasWZqORQjRAJ08Lvz7779j3LhxAIDx48fjjz/+0HAiQoimaLSIpaamIjg4GMOGDcOiRYvk5tXW1mLBggXg8/nw8fHBnj175OZ1794dwIM7hd+7d0+dsQkhWkSjx4Pt7OwQFRWF3Nxc1NTUyM2Li4tDU1MTcnJyUF5ejrCwMAwcOBACgQA2Njaoq6sDANTV1aFHjx6aiE8I0QIabYkFBATAz88PPXv2lJsuFAqRkZGBRYsWwdraGs7OzggNDUV6ejoAYNSoUThz5gwAIDs7G6NGjVJ7dkKIdtDKPrHi4mIwxuDk5CSb5uLigmvXrsl+d3BwwGuvvYbc3FxMmzZNU1EfSRtv964pLZerEKIKWnl6sVAohLW1tdw0GxsbNDQ0yB4vWbJE3bE4MZRLURRhSJfmEPXTypaYlZWVXMECHvR98Xg8DSXqnP9dftGs6Shagd4PogpaWcQGDBgAALLdR+DBiBl0nSYh5GEaLWISiQQikQgSiQRSqRQikQhisRhWVlYIDAxEQkIC6uvrUVhYiLS0NK3s+yKEaJZGi1hycjLc3NyQkpKCjIwMuLm5ISYmBgCwZs0amJqawtfXF+Hh4YiMjIRAINBkXEKIFtJox350dDSio6PbnWdjY4PExESVbVskEgEAioqKFFpeWFMGkVgCAGBmprh8+bLCz2mQWODy5cuPfAygw2XUsQ5tzdXeOh71N9B0Lm1Yp7aso711KvrZaflctnxOH8eIMcY6XEoPffvtt1i6dKmmYxBCHmPz5s2YOnXqY5cx2CJWXV2N06dPo3///rCwsNB0HEJIKyKRCGVlZfDx8YGtre1jlzXYIkYI0Q9aeYoFIYQoiooYIUSnUREjhOg0KmKEEJ1GRYwQotOoiBFCdBoVMUKITqMiRgjRaVTECCE6jYoYIUSnURHD428P97C8vDy8+OKLGDFiBKZNm4arV6+qLygUz/rnn39i7ty58PT0hKenJ9544w0UFxdrXc7WDh06BGdnZ+zfv1/1Af8fl5wikQjr1q3D2LFjMWrUKAQHB6O+vl7rch4/fhyTJk0Cn89HQEAADh48qJaMwONvw/gwpX2WGGFLlixh8+fPZ3V1dezq1avMy8uLZWVltVmuurqajR49mh0+fJiJRCK2a9cuNn78eCYSibQua1ZWFjt27Birra1lIpGIbdq0iU2cOFHrcraorq5mAQEBbPLkyWzfvn1amXP58uUsOjqa3b59mzU3N7P8/Hy1/e0VzVleXs5cXV1ZZmYmk0ql7Pz582zkyJHs8uXLasl54sQJ9uOPP7IPPviALVy48JHLKfOzZPBFrKGhgbm6urKCggLZtK1bt7Lo6Og2yx44cIAFBQXJHkulUjZu3Dh26tQpdUTllPVhd+7cYU5OTqy6ulqVERljncv53nvvsf3797OZM2eqrYhxyVlUVMRGjhzJ7t27p5ZsrXHJee7cOebl5SU3LTg4mB0+fFjVMeUkJiY+togp87Nk8LuTHd0errXCwkI888wzssdGRkZwdnZGYWGh1mV9WF5eHnr16tXmHp+qwDVnXl4eioqKEBoaqvJsrXHJefHiRfTv3x+ffvopvLy88MILL6htN41LzhEjRmDgwIH48ccfIZVKce7cOZSVlcHd3V0tWRWlzM+SVt6yTZ0UuT1c62Ufvtt49+7d211WFbhkba20tBRr167F6tWrVRlPhkvOpqYmxMXFIT4+HsbG6v1O5ZKzsrIShYWFeP7555GdnY2rV68iPDwcjo6O8PDw0JqcpqamCAoKwrJlyyASiWBkZIT3338f9vb2Ks3IlTI/SwbfEuNyezgrKyvU1dXJTauvr1fbreQ6cyu7iooKhIWFITIyEpMmTVJ1RADccu7YsQMeHh5wdXVVS7bWuOTs1q0bTExM8NZbb8Hc3Bxubm6YOHEiTp06pVU5T58+jU2bNmHXrl24dOkSjhw5guTkZGRlZak8JxfK/CwZfBHjcns4Jycn5Ofnyx4zxlBQUCDXzFclrreyq6ysxOzZs/HKK69gzpw56ogIgFvOX375BceOHYO3tze8vb1x/vx5fPTRR1ixYoVW5XR2dlZ5nkfhkrOgoACjRo0Cn8+HsbExhgwZAoFAgOzsbHXFVYgyP0sGX8S43B7O398fN27cwNGjR9HU1IQvvvgCAPDss89qXdaqqiq8/vrrmDp1Kt544w215OtMzm3btuH777/HkSNHcOTIEQwbNgzz589XSxHjktPd3R0ODg5ISUmBRCLBlStXcOLECTz//PNaldPNzQ3nz5/HxYsXAQDXr1/Hzz//DBcXF5XnBB59G8aHKfWzxPlQgB66d+8ei46OZiNHjmTe3t5s9+7dsnkjR45kv/32m+zx2bNn2eTJk9nw4cNZcHAwu3LlilZm3bZtG3NycmIjR46U+ykvL9eqnA9T59FJxrjl/Ouvv9j06dPZiBEjmL+/P0tLS9PKnPv27WMBAQFs5MiRTCAQsC1btrDm5ma15ExMTGROTk5yP8uXL283p7I+SzTGPiFEpxn87iQhRLdRESOE6DQqYoQQnUZFjBCi06iIEUJ0GhUxQohOoyJGCNFpVMQIITqNipge2blzJ3x8fMDn89VyYbIqREREYPv27ZqOoXTOzs7Izc3VdIwO8fl8/Prrr5qOwY0SrjQgHThw4ADz9vZm3t7e7MCBA3Lzdu7cyZYtW9blbVRWVjJnZ2d24cKFLq+Lq5kzZ7KtW7dyek5paSlzcnJixcXFKkqlGenp6czX17fNdCcnJ3bmzBkNJNJ/Bj+emKr9888/2LJlC9LT08EYw8svv4wJEybgySefRFFREQ4cOIC0tLQub6e0tBSMMQwdOlQJqQnRHbQ7qWLl5eUYMGAA+vfvDwcHBzg6OqK8vBwSiQTvvfceYmJi0L179w7XU1tbi5iYGIwbNw6enp6IiIjA9evXAQAHDx5EeHg4gAejLfD5/HbXcejQIYwbNw5fffUVnnvuOYwcORLvvfce6uvrsWbNGnh6esLHxwdff/11m+e0tm3bNkyfPh0AEBsbi3PnzmHnzp3g8/mybRcWFmL27Nnw8vLC6NGjERISgl9++UW2jhdffBEA8K9//Qt8Ph+xsbEAgFmzZiEhIUG2XFFRESIiIuDp6Ylx48YhNjZWbhyqWbNmYd26dViyZAlGjx4NgUCAffv2yebfvHkTkZGR8PDwwOjRo/Hiiy/i3Llz7b4/ZWVlcHZ2xjfffINJkyZh1KhRmD17NkpKSmTL/PDDDwgODoa7uzs8PT0xf/58lJaWAgDOnTuHNWvW4NatW7L34ttvv5V7La+++ir4fD6mTJkil6O5uRk7duxAYGAgRo8ejeDgYPz8889y+dLT0+Hv7w8+n4+3334bcXFxmDVrlmy+SCTC1q1b4efnB3d3d8yYMQNXrlxp83dLSkqCj48PPDw8EBsbC4lEIlvm4d3e8+fP4/XXX4enpyc8PDwwa9YsNDY2tvv+aYymm4L67u7du+zZZ59lJSUlrLi4mD377LPs7t27LCkpicXGxiq8nnnz5rGZM2eyW7duMaFQyNatW8fGjRvH6uvrGWMPRgRwcnJiYrH4ketIT09nzzzzDNu4cSNrbGxkJSUlzN3dnb3wwgvs5MmTTCKRsB9++IENHTqU3bx5U/ach3ePEhMT2auvvip73N7uZEFBATt9+jS7f/8+E4lELDExkY0aNYrduXOHMfbo3cnW66qrq2M+Pj5sw4YNTCgUsqqqKjZjxgz25ptvyi0/atQolpuby5qbm1lGRgZzcXGRrXfx4sVs1apVrLGxkTU3N7OioiJWUlLS7vvTkmnatGns5s2bTCgUspUrV7IXXnhB9r7+/PPPLD8/n0kkEvbPP/+wefPmsdDQULn3+FG7k1OmTGHFxcVMLBazdevWsfHjx8vm79y5k/n6+rJLly4xsVjMvv/+e+bq6souXbrEGGMsLy+Pubq6sqysLCaRSFhWVhZzc3NjM2fOlK1j+fLlbPbs2ayiooKJxWK2d+9e5uXlJbsvQGJiIhs6dCjbsWMHE4lE7Pr168zd3V1uNI7Wu72FhYVs+PDhLDU1lQmFQiYSidjZs2fVemMcRVBLTMV69OiBuLg4LFmyBEuXLkVcXBwqKirw3XffYfny5fjss88wc+ZMzJ8//5G3VLt16xZOnTqFVatWoVevXrC0tMSyZcvQ2NjIecROExMTLF68GBYWFnBwcIC7uzv69u0Lf39/mJiYYOLEibCwsMDly5e79LqdnJzg7e2Nbt26wdzcHNHR0TAyMsKFCxcUXkdWVhbEYjHeffddWFpaws7ODitXrkRmZiZu374tWy4gIABjx46FsbExAgMD0aNHD1l+c3Nz3LlzByUlJTAyMsKgQYPg4ODw2O1GRUWhb9++sLS0xIoVK1BcXIw///wTADBu3Di4uLjAxMQEtra2eOedd/Dnn38qdOu2luGsTU1NERoaips3b+LOnTsAgAMHDmDu3LlwdXWFqakpJk+eDF9fXxw4cAAAcOTIEUyYMAECgQAmJiYQCAQQCASyddfU1ODw4cOIjY1Fnz59YGpqipkzZ6J79+5yB3n69OmDuXPnwtzcHAMHDoSXl5ds7LGH7d+/H2PHjsWMGTNgaWkJc3NzeHp6wtzcvMPXqk7UJ6YGEyZMwIQJEwA8GFM+NDQUa9euxfnz55GTk4M9e/bgzJkzWLlypdyuUIvKykoAkPvwmZmZwd7eHjdv3uSUpWfPnjAzM5M9trS0bLM7a2lp2eX7Bty8eRObN2/G+fPnUVtbC2NjY9TX1+Off/5ReB0VFRXo168fTE3/92/69NNPy+b16tULAGBnZyf3vNbDOS9btgzJycl4++23UVtbi/Hjx2PJkiV46qmnHrnd/v37y363trZGz549UVFRAeDBTU22b9+OoqIiCIVC2XLV1dVtxsF/WOuclpaWAICGhgY89dRTqKyslL22Fo6OjigqKgLwYJDLh0dytbe3R01NDQDIdnkfvtmKWCxGVVVVuxmA9oe+blFWVoZBgwY99jVpA2qJqVlSUhI8PT3h7u6OK1eugM/nw9TUFF5eXnL9F6316dMHAGR9L8CDETRv3ryJfv36qTQvj8fD/fv35abdunVL7rGRkVGb58XExEAqlSItLQ1//PEHfvvtN7kPuSI3Benbty8qKirk+mxa3oO+ffsqlL9nz55YuXIlTpw4gSNHjqCsrAwbNmx47HPKy8tlvzc0NKCmpgZ9+vRBU1MT5s2bB19fX2RkZOCPP/5AamoqgAfDKyv6utrTp08fub434EFhanmdvXv3bvOF1TpnS1H+9ttvce7cOdnPhQsXOj2yr729vVpvuNxZVMTU6MKFCzh16hQWL14M4MHY6WfPnoVQKERmZmabb+IWdnZ2EAgE2LhxI+7cuYPGxkZ89NFHMDMzk9ulUIWhQ4eioaEBx48fh1Qqxa+//oqMjAy5ZXr16tXmn72urg5WVlawsbGBUCjEli1b5Foutra2MDY2xt9///3IbY8fPx4mJibYunUrGhsbcfv2baxfvx7PPfecrBXWkWPHjqGkpARSqRQ8Hg/m5uZyLbv2JCcno7KyEvfv30d8fDyefvpp8Pl8iMViiEQi9OjRA9bW1qiqqsLHH38s99ynnnoKNTU1shaSokJCQrBr1y7k5+dDIpHg+PHjyM7ORkhICIAHB0AyMzORk5OD5uZmZGdny42bb29vDz8/P8TFxcmKW319PX7++ec2XzqKmj59Os6cOYP9+/ejsbERYrEYeXl5aGpq6tT6VIWKmJqIRCKsWrUKH374ISwsLAAAfn5+eOaZZyAQCLBt2zbExcU98vmbNm2Cvb09goKCIBAIcO3aNezZs6fDXZiucnBwQExMDOLj4zFmzBgcOHAAwcHBcsvMmTMHf//9N9zd3TFmzBgAwOrVq1FQUAAPDw9MnjwZvXv3lrUogQd3D1q0aBFWr16NMWPG4P3332+zbWtra+zevRsFBQUQCAQIDg6Go6Mj4uPjFc5/9epVzJ49G6NGjYKfnx9sbGywfPnyxz7n5Zdfxpw5c/Dss8/i77//RkpKCkxNTcHj8bBu3TokJyeDz+cjMjISEydOlHuul5cXJkyYgIkTJ2LMmDH47rvvFMo5Z84czJgxA++88w48PDywY8cObNu2DcOHDwcAeHh4YM2aNfjggw9kf4cpU6bI9U9t2bIFrq6uCA8PB5/Px8SJE3Hw4EFZK5ErJycn7N69G8eOHYOvry+8vb2xfft2SKXSTq1PVWh4akL+X1lZGSZMmICTJ0/C0dFR03E6FBUVhaeeeuqxX36GgFpihOiIEydOoL6+HmKxGMePH0dWVhamTJmi6VgaR0cnCdERp06dQkxMDMRiMfr164f169fD3d1d07E0jnYnCSE6jXYnCSE6jYoYIUSnUREjhOg0KmKEEJ1GRYwQotOoiBFCdBoVMUKITqMiRgjRaf8HdyhBWtRLZcYAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 300x200 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.set(context='paper', style='white')\n",
    "plt.figure(figsize=(3,2))\n",
    "plt.hist(path_enrich['pct_pathogenic'], bins=50)\n",
    "plt.xlabel('% of mutations pathogenic')\n",
    "plt.ylabel('Number of clusters')\n",
    "plt.yscale('log')\n",
    "plt.savefig('./figures/cluster_pct_pathogenic.svg', dpi=300, bbox_inches='tight', format='svg')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13569"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(path_enrich['pct_pathogenic'] > 0).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2056, 23075, 43)"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig = path_enrich[(path_enrich['pval_corr'] < 0.05) & (path_enrich['n_mut'] > 1)]\n",
    "len(sig), len(path_enrich), len(sig[sig['n_mut'] > 5])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({2: 1406,\n",
       "         3: 418,\n",
       "         4: 128,\n",
       "         5: 61,\n",
       "         6: 21,\n",
       "         7: 10,\n",
       "         8: 3,\n",
       "         9: 2,\n",
       "         31: 1,\n",
       "         85: 1,\n",
       "         20: 1,\n",
       "         35: 1,\n",
       "         65: 1,\n",
       "         32: 1,\n",
       "         19: 1})"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col.Counter(sig['n_mut'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathogenic_clusters_5 = sig[sig['n_mut'] >= 5].index.tolist()\n",
    "pathogenic_clusters = sig.index.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pval</th>\n",
       "      <th>pval_corr</th>\n",
       "      <th>mutations</th>\n",
       "      <th>pct_pathogenic</th>\n",
       "      <th>n_mut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6044</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>0.964706</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12238</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>0.984615</td>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26414</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>0.971429</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6563</th>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.040722</td>\n",
       "      <td>[pathogenic, pathogenic, benign, pathogenic, p...</td>\n",
       "      <td>0.84375</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11387</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009237</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>0.870968</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41532</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18464</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37660</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20221</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41913</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic, pathogeni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>104 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           pval pval_corr                                          mutations  \\\n",
       "6044        0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "12238       0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "26414       0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "6563   0.000002  0.040722  [pathogenic, pathogenic, benign, pathogenic, p...   \n",
       "11387       0.0  0.009237  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "...         ...       ...                                                ...   \n",
       "41532       0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "18464       0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "37660       0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "20221       0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "41913       0.0       0.0  [pathogenic, pathogenic, pathogenic, pathogeni...   \n",
       "\n",
       "      pct_pathogenic  n_mut  \n",
       "6044        0.964706     85  \n",
       "12238       0.984615     65  \n",
       "26414       0.971429     35  \n",
       "6563         0.84375     32  \n",
       "11387       0.870968     31  \n",
       "...              ...    ...  \n",
       "41532            1.0      5  \n",
       "18464            1.0      5  \n",
       "37660            1.0      5  \n",
       "20221            1.0      5  \n",
       "41913            1.0      5  \n",
       "\n",
       "[104 rows x 5 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig[sig['n_mut'] >= 5].sort_values('n_mut', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_id = 23310\n",
    "cluster_mut_df = []\n",
    "for uniprot_mut, lab in list(zip(cluster_refs[cluster_id], cluster_mutations[cluster_id])):\n",
    "    # print('\\n' + uniprot + '\\n' + '------')\n",
    "    uniprot, resid = uniprot_mut.split('_')\n",
    "    loc = int(resid[1:])\n",
    "    muts = mut_data.loc[uniprot_rows[uniprot]]\n",
    "    muts = muts[muts.pos == loc]\n",
    "    cluster_mut_df.append(muts)\n",
    "cluster_mut_df = pd.concat(cluster_mut_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_mut_df['variant'] = cluster_mut_df['wt'].apply(lambda x: aa_to_letter(x.upper())) + cluster_mut_df['pos'].astype(str) + cluster_mut_df['mut'].apply(lambda x: aa_to_letter(x.upper()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_variants_df = pd.DataFrame([cluster_mut_df.groupby(['uniprot'])['variant'].apply(list), cluster_mut_df.groupby(['uniprot'])['label'].apply(list)]).T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>uniprot</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P04629</th>\n",
       "      <td>[Y359C]</td>\n",
       "      <td>[pathogenic]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P11362</th>\n",
       "      <td>[Y99C, Y228D, Y339C]</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P21802</th>\n",
       "      <td>[Y105C, Y340H, Y340C]</td>\n",
       "      <td>[pathogenic, pathogenic, pathogenic]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q0ZGT2</th>\n",
       "      <td>[Y652C]</td>\n",
       "      <td>[pathogenic]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q14896</th>\n",
       "      <td>[Y237S]</td>\n",
       "      <td>[pathogenic]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q86TC9</th>\n",
       "      <td>[Y339F]</td>\n",
       "      <td>[pathogenic]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       variant                                 label\n",
       "uniprot                                                             \n",
       "P04629                 [Y359C]                          [pathogenic]\n",
       "P11362    [Y99C, Y228D, Y339C]  [pathogenic, pathogenic, pathogenic]\n",
       "P21802   [Y105C, Y340H, Y340C]  [pathogenic, pathogenic, pathogenic]\n",
       "Q0ZGT2                 [Y652C]                          [pathogenic]\n",
       "Q14896                 [Y237S]                          [pathogenic]\n",
       "Q86TC9                 [Y339F]                          [pathogenic]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_variants_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_variants_df.to_csv(f'data/cluster{cluster_id}_variants.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      uniprot   wt  pos  mut       label  \\\n",
      "22379  P22413  Asp  804  His  pathogenic   \n",
      "\n",
      "                                                 disease variant  \n",
      "22379  Arterial calcification of infancy, generalized...   D804H  \n",
      "      uniprot   wt  pos  mut       label  \\\n",
      "16477  P49747  Asp  290  Asn  pathogenic   \n",
      "16511  P49747  Asp  290  Gly  pathogenic   \n",
      "16514  P49747  Asp  317  Gly  pathogenic   \n",
      "16515  P49747  Asp  326  Gly  pathogenic   \n",
      "16516  P49747  Asp  326  Tyr  pathogenic   \n",
      "16480  P49747  Asp  342  Tyr  pathogenic   \n",
      "16482  P49747  Asp  361  Val  pathogenic   \n",
      "16483  P49747  Asp  361  Tyr  pathogenic   \n",
      "16519  P49747  Asp  374  Asn  pathogenic   \n",
      "16520  P49747  Asp  376  Asn  pathogenic   \n",
      "16521  P49747  Asp  378  Val  pathogenic   \n",
      "16522  P49747  Asp  385  Asn  pathogenic   \n",
      "16523  P49747  Asp  385  Tyr  pathogenic   \n",
      "16525  P49747  Asp  397  His  pathogenic   \n",
      "16486  P49747  Asp  408  Tyr  pathogenic   \n",
      "16507  P49747  Asp  420  Ala  pathogenic   \n",
      "16530  P49747  Asp  446  Asn  pathogenic   \n",
      "16492  P49747  Asp  473  Gly  pathogenic   \n",
      "16532  P49747  Asp  473  His  pathogenic   \n",
      "16533  P49747  Asp  475  Asn  pathogenic   \n",
      "16493  P49747  Asp  482  Gly  pathogenic   \n",
      "16535  P49747  Asp  507  Gly  pathogenic   \n",
      "16536  P49747  Asp  511  Gly  pathogenic   \n",
      "\n",
      "                                                 disease variant  \n",
      "16477          Pseudoachondroplasia (PSACH) [MIM:177170]   D290N  \n",
      "16511          Pseudoachondroplasia (PSACH) [MIM:177170]   D290G  \n",
      "16514  Multiple epiphyseal dysplasia 1 (EDM1) [MIM:13...   D317G  \n",
      "16515  Multiple epiphyseal dysplasia 1 (EDM1) [MIM:13...   D326G  \n",
      "16516          Pseudoachondroplasia (PSACH) [MIM:177170]   D326Y  \n",
      "16480  Multiple epiphyseal dysplasia 1 (EDM1) [MIM:13...   D342Y  \n",
      "16482  Multiple epiphyseal dysplasia 1 (EDM1) [MIM:13...   D361V  \n",
      "16483  Multiple epiphyseal dysplasia 1 (EDM1) [MIM:13...   D361Y  \n",
      "16519  Multiple epiphyseal dysplasia 1 (EDM1) [MIM:13...   D374N  \n",
      "16520  Multiple epiphyseal dysplasia 1 (EDM1) [MIM:13...   D376N  \n",
      "16521          Pseudoachondroplasia (PSACH) [MIM:177170]   D378V  \n",
      "16522  Multiple epiphyseal dysplasia 1 (EDM1) [MIM:13...   D385N  \n",
      "16523  Multiple epiphyseal dysplasia 1 (EDM1) [MIM:13...   D385Y  \n",
      "16525  Multiple epiphyseal dysplasia 1 (EDM1) [MIM:13...   D397H  \n",
      "16486  Multiple epiphyseal dysplasia 1 (EDM1) [MIM:13...   D408Y  \n",
      "16507  Multiple epiphyseal dysplasia 1 (EDM1) [MIM:13...   D420A  \n",
      "16530          Pseudoachondroplasia (PSACH) [MIM:177170]   D446N  \n",
      "16492          Pseudoachondroplasia (PSACH) [MIM:177170]   D473G  \n",
      "16532          Pseudoachondroplasia (PSACH) [MIM:177170]   D473H  \n",
      "16533          Pseudoachondroplasia (PSACH) [MIM:177170]   D475N  \n",
      "16493          Pseudoachondroplasia (PSACH) [MIM:177170]   D482G  \n",
      "16535          Pseudoachondroplasia (PSACH) [MIM:177170]   D507G  \n",
      "16536          Pseudoachondroplasia (PSACH) [MIM:177170]   D511G  \n"
     ]
    }
   ],
   "source": [
    "for u, df in cluster_mut_df.groupby('uniprot'):\n",
    "    print(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "uniprot\n",
       "O75096     1\n",
       "P00740     3\n",
       "P01130     9\n",
       "P07225     4\n",
       "P07911     5\n",
       "P08709     2\n",
       "P23352     3\n",
       "P24043     1\n",
       "P35555    56\n",
       "P35556    12\n",
       "P46531     1\n",
       "P78504     1\n",
       "P82279     5\n",
       "Q5IJ48     1\n",
       "Q5T1H1     1\n",
       "Q6UXH8     1\n",
       "Q92832     1\n",
       "Q9HC23     1\n",
       "Q9NR61     1\n",
       "Name: variant, dtype: int64"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_variants_df['variant'].apply(len)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 21834/21834 [3:01:51<00:00,  2.00it/s]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9703 proteins of 21834 contain pathogenic clusters\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "count_path = 0\n",
    "count_path_5 = 0\n",
    "pathogenic_residues = []\n",
    "for item in tqdm(human_proteome):\n",
    "    if item is None:\n",
    "        continue\n",
    "    afid, _ = pdb_from_fname(item['id'])\n",
    "    \n",
    "    high_conf_idx = np.array(item['confidence']) >= 70\n",
    "    if np.sum(high_conf_idx) == 0:\n",
    "        continue\n",
    "    af_resids = np.array(item['resids'])[high_conf_idx]\n",
    "    embeddings = item['embeddings'][high_conf_idx]\n",
    "    \n",
    "    clusters = clusterer.predict(embeddings)\n",
    "    # if np.any(np.isin(clusters, pathogenic_clusters)):\n",
    "    #     count_path += 1\n",
    "    pathogenic_pos = np.isin(clusters, pathogenic_clusters_5)\n",
    "    pathogenic_resids = af_resids[pathogenic_pos]\n",
    "    pathogenic_residues.extend([[afid, r] for r in pathogenic_resids])\n",
    "    if np.any(pathogenic_pos):\n",
    "        count_path_5 += 1\n",
    "print(f'{count_path_5} proteins of {len(human_proteome)} contain pathogenic clusters')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "39383"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pathogenic_residues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.DataFrame(pathogenic_residues, columns=['uniprot', 'resid']).to_csv('data/human_possible_pathogenic_resids.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
